dvc/utils/serialize/_py.py

```diff
@@ -0,0 +1,96 @@
+import ast
+
+from funcy import reraise
+
+from ._common import ParseError, _load_data
+
+
+class PYFileCorruptedError(ParseError):
+    def __init__(self, path, message="PY file structure is corrupted"):
+        super().__init__(path, message)
+
+
+def load_py(path, tree=None):
+    return _load_data(path, parser=parse_py, tree=tree)
+
+
+def parse_py(text, path):
+    """Parses text from .py file into Python structure."""
+    with reraise(SyntaxError, PYFileCorruptedError(path)):
+        tree = ast.parse(text, filename=path)
+
+    result = ast_tree_to_dict(tree)
+    return result
+
+
+def ast_tree_to_dict(tree):
+    """Parses ast trees to dict."""
+    result = {}
+    for _body in tree.body:
+        try:
+            if isinstance(_body, ast.Assign) or isinstance(
+                _body, ast.AnnAssign
+            ):
+                result.update(ast_assign_to_dict(_body))
+            elif isinstance(_body, ast.ClassDef):
+                result.update({_body.name: ast_tree_to_dict(_body)})
+            elif (
+                isinstance(_body, ast.FunctionDef) and _body.name == "__init__"
+            ):
+                result.update(ast_tree_to_dict(_body))
```

Comment:
Looks like this reads from variable instead of `self.*` right?

```python
class Klass:
    def __init__(self):
       self.a = 3
```

I'd expect `Klass.a` be 3 here. If it's difficult, we could remove this functionality.

Reply:
Right, `Klass.a` be 3. It's works. Added to tests.

Reply:
Ahh, I thought it didn't. Now, I see how it works, but might be confusing to the users, especially in the following scenarios:
1.
```python
x = 3
config.x = 3
```
`parse_py` gives me result `{"x": 3}`.

2. 
```python
class Config:

   def __init__(self):
      self.container.value = 5
       self.value = 1
       value = 4  # used for something else
```

`parse_py` gives me result `{"Config": {"value": "4"}}`, instead of expected "1" (neither 4 or 5 is expected).

Reply:
Yeah, added a flag to read only self parameters

Reply:
Hi @aandrusenko, this problem still exists, right? See the other two scenarios with `self.container.value` and `container.x` above.

Reply:
We could either ignore these, or return a dict containing ```json
{"Config": {"container": {"value": 5}}}
```

Reply:
Fixed, reads only `self.value` without any implementations

Reply:
Hi, this still fails for non-class attr assignment, right? 
```python
>>> parse_py("x = 4; config.x = 3", "file name does not matter")
{'x': 3}
```

Reply:
@skshetry, fixed, but I have no more ideas how to ignore `self.x` outside the class, `ast` has no more functionality. In my opinion, it is strange to meet `self.x` outside the class, because `self` is not defined and params.py cannot even be imported